lib/gap_intelligence/client/requestable.rb

```diff
@@ -8,21 +8,21 @@ module Requestable
     # @yield [req] The Faraday request
     def perform_request(method, path, options = {}, &block)
       record_class = options.delete(:record_class)
+      response_processor = options.delete(:response_processor)
```

Comment:
@gtimti according to existing logic if parsed response is `Array` then we process it as a set of records and create new `RecordSet` object, if parsed response is `Hash` then we process it as a single `Record`. In the case of MerchantPricingTrend response is indeed an Array of trend data, however it will be *expensive to process every single row of MPT data as a separate instance as this will instantiate thousands of objects, instead you can pass a `response_processor` and control either you want response to processed as single object (Record) or a set of objects of `record_class`. So this option is used in the logic on line 23 and 26